### ISSUE TITLE
Latest version not reporting/suggesting changes on disabled/value attributes

### ISSUE THREAD

**Issue by ricardozv28** [testing-library/eslint-plugin-jest-dom#203]
- `eslint-plugin-jest-dom` version: 3.9.2
- `node` version: 14.17.6
- `npm` version: 6.14.15

Relevant code or config

```javascript
    const button = getByRole('button', { name: 'My Button' });
    expect(button.disabled).toBe(true);

    const input = getByLabelText('My Input');
    expect(input.value).toBe('value');
```

What you did:
run eslint using `npm run eslint`

What happened:
eslint is not reporting suggested changes when using `.disabled` or `.value`

Reproduction repository:
https://github.com/ricardozv28/eslint-plugin-jest-dom-bug

Problem description:
When using the latest versions of the package, eslint won't report on `.disabled` or `.value` (I haven't tested the other rules but those are the ones that I caught when adding the plugin to an existing repo at work) 

Here is a screenshot of the output of the reproducible projet, it is just showing the ones from eslint-plugin-testing-librarry
![image](https://user-images.githubusercontent.com/23007152/134742042-ae46ed2f-c326-4b43-a991-0bf7bc2e19ad.png)

downgrading to version 3.6.3 make the errors appear
![image](https://user-images.githubusercontent.com/23007152/134741778-231c6a82-d047-4eaf-9c7b-2f42bb96a89c.png)

version 3.6.4 only reports the `toHaveValue`
![image](https://user-images.githubusercontent.com/23007152/134741934-03116bcb-858f-40c5-921d-e5b8f2294564.png)

Suggested solution:

**Comment by G-Rath** [comment]
It looks like at least for `prefer-to-have-value` we currently only support `*ByRole`:

https://github.com/testing-library/eslint-plugin-jest-dom/blob/ee6af993112ac73a4053294e2e55e7ff2ba926f9/src/rules/prefer-to-have-value.js#L35-L35

@benmonro @MichaelDeBoey do either of you know if there's a reason for that? My understanding is that all the queries return elements, so we should be ok to apply this rule to any of the query methods.

(I'm guessing this is probably also the case for `prefer-enabled-disabled` but have not found where it's actually doing the check yet)

**Comment by MichaelDeBoey** [comment]
@G-Rath It seems like this was first introduced by @juzerzarif in #122

**Comment by juzerzarif** [comment]
I believe it was because `ByRole` queries are the only ones where we can be definitely sure that the element returned is an input element that has a `value` attribute (via the role specified). It was a while ago though so @benmonro can correct me if I'm wrong.

**Comment by G-Rath** [comment]
@juzerzarif that makes sense - in that case if we want to keep it like that we should probably stick a note in the doc; could also be worth considering if we want to provide a `loose` option for people who'd prefer to risk the false positives.

(I imagine that's the case with a number of rules - could be interesting to see if optionally using type-checking if available could  let the rules be smarter)

**Comment by benmonro** [comment]
Yeah adding a note to the docs makes sense to me
